cudamapper/src/index_cache.cu

```diff
@@ -0,0 +1,257 @@
+/*
+* Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#include "index_cache.cuh"
+
+#include "index_host_copy.cu"
+
+#include <claragenomics/cudamapper/index.hpp>
+#include <claragenomics/io/fasta_parser.hpp>
+
+namespace claragenomics
+{
+namespace cudamapper
+{
+
+IndexDescriptor::IndexDescriptor(read_id_t first_read, read_id_t number_of_reads)
+    : first_read_(first_read)
+    , number_of_reads_(number_of_reads)
+    , hash_(0)
+{
+    generate_hash();
+}
+
+read_id_t IndexDescriptor::first_read() const
+{
+    return first_read_;
+}
+
+read_id_t IndexDescriptor::number_of_reads() const
+{
+    return number_of_reads_;
+}
+
+std::size_t IndexDescriptor::get_hash() const
+{
+    return hash_;
+}
+
+void IndexDescriptor::generate_hash()
+{
+    // populate lower half of hash with one value, upper half with the other
+    std::size_t element_mask = 0;
+    std::uint32_t shift_bits = 0;
+    if (sizeof(std::size_t) == 4)
+    {
+        // 16 bytes per value
+        element_mask = 0xFFFF;
+        shift_bits   = 16;
+    }
+    else if (sizeof(std::size_t) == 8)
```

Comment:
I was thinking a lot about it. I also had a version with automatic deduction of these values similar to the one you have, but then ditched it because of readability.
Anyway, I agree it's safe to assume that it's only going to be used on 64-bit machines. I'll hardcore the values and add the `static_assert` (and also adapt the tests)

Reply:
done